### PR TITLE
Adding class label indicators validation.

### LINKED ISSUE #197: ClassLabelIndicatorsFromIntLabels should error when given labels outside [0, numClasses)
It's not immediately obvious that `ClassLabelIndicatorsFromIntLabels` wants contiguous class labels, and a naive user might give it labels like {-1, +1} or simply unique keys. Throwing an error here would be great.

### PR DESCRIPTION
**Pull request by etrain**
Closes #197 

Added unit tests.

### CODE REVIEW

File: src/main/scala/nodes/util/ClassLabelIndicators.scala [hunk 1]
```diff
@@ -21,12 +30,21 @@ case class ClassLabelIndicatorsFromIntLabels(numClasses: Int)
...
  * @param numClasses
  */
-case class ClassLabelIndicatorsFromIntArrayLabels(numClasses: Int)
+case class ClassLabelIndicatorsFromIntArrayLabels(numClasses: Int, validate: Boolean = false)
     extends Transformer[Array[Int], DenseVector[Double]] {
 
+  assert(numClasses > 1, "numClasses must be > 1.")
+
   def apply(in: Array[Int]): DenseVector[Double] = {
+    if(validate && (in.max > numClasses || in.min < 0)) {
```
Comment (shivaram) [inline src/main/scala/nodes/util/ClassLabelIndicators.scala]: I think this should be `>= numClasses` ?

File: src/main/scala/nodes/util/ClassLabelIndicators.scala [hunk 2]
```diff
@@ -7,12 +7,21 @@ import workflow.Transformer
 
 /**
  * Given a class label, returns a binary vector that indicates when that class is present.
+ *
+ * Expects labels in the range [1, numClasses).
```
Comment (shivaram) [inline src/main/scala/nodes/util/ClassLabelIndicators.scala]: This should be `[0, numClasses)` ?

**Comment by etrain** [comment]
Good catches.

**Comment by shivaram** [comment]
LGTM

**Pull request merged**